_posts/2015-07-20-CocoaPods-0.38.markdown

```diff
@@ -0,0 +1,140 @@
+---
+layout: post
+title:  "CocoaPods 0.38 - watchOS and target de-duplication"
+author: marius
+categories: cocoapods releases
+---
+
+TL;DR: _CocoaPods 0.38_ has been released, with watchOS 2 support, target de-duplication and Stats for everyone.
+
+<!-- more -->
+
+CocoaPods 0.38 has been released.
+This is the first update since WWDC and it brings support for Apple's latest releases including Xcode 7 and watchOS.
+It also includes lots of improvements that make building Pods easier.
+Furthermore it rolls out [Stats](http://blog.cocoapods.org/Stats/) for everyone.
+
+
+### watchOS
+
+The update allows you to use Pods with watchOS.
+As a Pod Author, you have to specify the [deployment target
+for watchOS](https://guides.cocoapods.org/syntax/podspec.html#watchos) in the Podspec, as seen below:
+
+```ruby
+Pod::Spec.new do |s|
+  # …
+  s.watchos.deployment_target = '2.0'
+end
+```
+
+What this means in practice, is that not all Pods are automatically available by default.
+Any Pod that can support watchOS needs to explicitly add support by adding `s.watchos.deployment_target`.
+It is an entirely new platform.
+
+### Target Deduplication
+
+CocoaPods now de-duplicates Pod targets, which means that it recognises when a dependency is used multiple times across different user targets but only needs to be built once.
+But sometimes this is not possible and they still need to be generated in different variants and thus build multiple times.
+The targets in `Pods.xcodeproj` need to be duplicated when one of the following conditions applies:
+
+* They are used on different platforms.
+* They are used with differents sets of subspecs.
+* They have any dependency which needs to be duplicated.
```

Comment:
I don't think that's obvious. It made obviously also a difference in the implementation. It will help to find the cause if you would expect that something is de-duplicated, but find the pod target duplicated.